program/src/guards/freeze_sol_payment.rs

```diff
@@ -0,0 +1,493 @@
+        ];
+
+        let (pda, _) = Pubkey::find_program_address(&seeds, &crate::ID);
+        assert_keys_equal(freeze_pda.key, &pda)?;
+        if freeze_pda.data_is_empty() {
+            return err!(CandyGuardError::FreezeNotInitialized);
+        }
+
+        let nft_ata = Self::get_account_info(ctx, index + 1)?;
+        evaluation_context.account_cursor += 1;
+        assert_is_ata(nft_ata, ctx.accounts.payer.key, ctx.accounts.nft_mint.key)?;
+
+        evaluation_context
+            .indices
+            .insert("freeze_sol_payment", index);
+
+        if ctx.accounts.payer.lamports() < self.lamports {
+            msg!(
+                "Require {} lamports, accounts has {} lamports",
+                self.lamports,
+                ctx.accounts.payer.lamports(),
+            );
+            return err!(CandyGuardError::NotEnoughSOL);
+        }
+
+        Ok(())
+    }
+
+    fn pre_actions<'info>(
+        &self,
+        ctx: &Context<'_, '_, '_, 'info, Mint<'info>>,
+        _mint_args: &[u8],
+        _guard_set: &GuardSet,
+        evaluation_context: &mut EvaluationContext,
+    ) -> Result<()> {
+        let freeze_pda =
+            Self::get_account_info(ctx, evaluation_context.indices["freeze_sol_payment"])?;
+
+        invoke(
+            &system_instruction::transfer(
+                &ctx.accounts.payer.key(),
+                &freeze_pda.key(),
+                self.lamports,
+            ),
+            &[
+                ctx.accounts.payer.to_account_info(),
+                freeze_pda.to_account_info(),
+                ctx.accounts.system_program.to_account_info(),
+            ],
+        )?;
+
+        Ok(())
+    }
+
+    fn post_actions<'info>(
+        &self,
+        ctx: &Context<'_, '_, '_, 'info, Mint<'info>>,
+        _mint_args: &[u8],
+        _guard_set: &GuardSet,
+        evaluation_context: &mut EvaluationContext,
+    ) -> Result<()> {
+        msg!("Post actions");
+        let index = evaluation_context.indices["freeze_sol_payment"];
+        let freeze_pda = Self::get_account_info(ctx, index)?;
+
+        let mut freeze_escrow: Account<FreezeEscrow> = Account::try_from(freeze_pda)?;
+        freeze_escrow.frozen_count += 1;
+
+        if freeze_escrow.first_mint_time.is_none() {
+            let clock = Clock::get()?;
+            freeze_escrow.first_mint_time = Some(clock.unix_timestamp);
+        }
+
+        freeze_escrow.exit(&crate::ID)?;
+
+        let candy_guard_key = &ctx.accounts.candy_guard.key();
+        let candy_machine_key = &ctx.accounts.candy_machine.key();
+        let payer = &ctx.accounts.payer;
+
+        let seeds = [
+            b"freeze_sol_payment".as_ref(),
+            self.destination.as_ref(),
+            candy_guard_key.as_ref(),
+            candy_machine_key.as_ref(),
+        ];
+
+        let (_, bump) = Pubkey::find_program_address(&seeds, &crate::ID);
+        let signer = [seeds[0], seeds[1], seeds[2], seeds[3], &[bump]];
```

Comment:
Is there a more idiomatic way to do this?

Reply:
Changed to:
```
let signer = [
            FreezeEscrow::PREFIX_SEED,
            self.destination.as_ref(),
            candy_guard_key.as_ref(),
            candy_machine_key.as_ref(),
            &[bump],
        ];
```